### PR TITLE
[Backport of 43846 to 14_0_X] Primary vertex reconstruction: update configuration handling

### PR DESCRIPTION
**Pull request by fabiocos**
#### PR description:

Verbatim backport of #43846 into the 14_0_X branch, where the initial update of primary vertex reconstruction #43592 was merged.

#### PR validation:

Code cherry-picked with no conflict and compiled without problems from the original branch successfully integrated in 14_1_X .

### CODE REVIEW

**Comment by fabiocos** [comment]
please test

**Comment by jfernan2** [comment]
+1

**Comment by tjavaid** [comment]
backport of #43846

**Comment by perrotta** [comment]
+1
- Verbatim backport of https://github.com/cms-sw/cmssw/pull/43846, which is already merged in master

**Comment by rappoccio** [comment]
+1

File: RecoVertex/PrimaryVertexProducer/plugins/PrimaryVertexProducer.cc [hunk 1]
```diff
@@ -451,7 +457,7 @@ void PrimaryVertexProducer::fillDescriptions(edm::ConfigurationDescriptions& des
   desc.add<bool>("useMVACut", false);
   desc.add<double>("minTrackTimeQuality", 0.8);
 
-  descriptions.add("primaryVertexProducer", desc);
+  descriptions.addDefault(desc);
```
Comment (Martin-Grunewald) [inline RecoVertex/PrimaryVertexProducer/plugins/PrimaryVertexProducer.cc:460]: Same here - ConfDb parsing needs an explicit cfi file. I assume you want descriptions.addWithDefaultLabel ?!

Comment (Martin-Grunewald) [reply RecoVertex/PrimaryVertexProducer/plugins/PrimaryVertexProducer.cc:460]: Fixed by #44288

**Pull request merged**